Analysis/Stan/mapping_model_generate.stan

```diff
@@ -602,32 +579,43 @@ generated quantities {
       // rigth-censored observations
       for(i in 1:M_right){
         real lpmf;
-        real dummy_right;
-        int j = ind_right[i];
-        
-        dummy_right = y[j] + exp(raw_dummy_right[i]);
-        
         if (obs_model == 1) {
           // Poisson likelihood
-          lpmf = poisson_cont_lpdf(dummy_right | modeled_cases[j]);
+          lpmf = poisson_lpmf(y[ind_right[i]] | modeled_cases[ind_right[i]]);
         } else if (obs_model == 2) {
           // Quasi-poisson likelihood
-          lpmf = neg_binomial_2_cont_lpdf(dummy_right | modeled_cases[j], od_param[map_obs_admin_lev[j]] * modeled_cases[j]);
+          lpmf = neg_binomial_2_lpmf(y[ind_right[i]] | modeled_cases[ind_right[i]], od_param[map_obs_admin_lev[ind_right[i]]] * modeled_cases[ind_right[i]]);
         } else {
           // Neg-binom likelihood
-          lpmf = neg_binomial_2_cont_lpdf(dummy_right | modeled_cases[j], od_param[map_obs_admin_lev[j]]);
+          lpmf = neg_binomial_2_lpmf(y[ind_right[i]] | modeled_cases[ind_right[i]], od_param[map_obs_admin_lev[ind_right[i]]]);
         }
         
-        // Add constrained likelihood and log-jacobian det for transformation
-        log_lik[j] = lpmf;
+        // heuristic condition to only use the PMF if Prob(Y>y| modeled_cases) ~ 0
+        if ((y[ind_right[i]] < modeled_cases[ind_right[i]]) || ((y[ind_right[i]] > modeled_cases[ind_right[i]]) && (lpmf > -35))) {
+          array[2] real lls;
+          if (obs_model == 1) {
+            // Poisson likelihood
+            lls[1] = poisson_lccdf(y[ind_right[i]] | modeled_cases[ind_right[i]]);
+          } else if (obs_model == 2) {
+            // Quasi-poisson likelihood
+            lls[1] = neg_binomial_2_lccdf(y[ind_right[i]] | modeled_cases[ind_right[i]], od_param[map_obs_admin_lev[ind_right[i]]] * modeled_cases[ind_right[i]]);
```

Comment:
@javierps I'm confused about why we are adding the old censoring methodology to the genquant section here...

Reply:
Good point, things are updated to follow the inference code.